internal/types/compare_order.go

```diff
@@ -110,7 +137,19 @@ func CompareOrder(a, b any, order SortType) CompareResult {
 		return result
 	}
 
-	return Compare(a, b)
+	c := &comparer{
+		compareTypeOrder: compareType,
```

Comment:
Within array and within document, we must also check for different number type, since we pass this `compareType` around.